engine/src/main/java/org/kigalisim/engine/SingleThreadEngine.java

```diff
@@ -538,7 +619,8 @@ public void cap(String stream, EngineNumber amount, YearMatcher yearMatcher,
     if (amount.hasEquipmentUnits()) {
       // For equipment units, convert to kg and add recharge volume on top
       EngineNumber amountInKg = unitConverter.convert(amount, "kg");
-      EngineNumber rechargeVolume = calculateRechargeVolume();
+      EngineNumber rechargeVolume = RechargeVolumeCalculator.calculateRechargeVolume(
+          this.scope, this.stateGetter, this.streamKeeper, this);
```

Comment:
Please use line continuation style here where there is one parameter per line with a comma at the end.